kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/serialization/debug/SerializationDebugRequestHandler.java

```diff
@@ -0,0 +1,404 @@
+                                .getPushMode().isEnabled();
+                        SessionDebugNotifier debugNotifier = ui.getChildren()
+                                .filter(SessionDebugNotifier.class::isInstance)
+                                .map(SessionDebugNotifier.class::cast).findAny()
+                                .orElseGet(() -> {
+                                    SessionDebugNotifier notifier = new SessionDebugNotifier();
+                                    ui.add(notifier);
+                                    return notifier;
+                                });
+                        if (pushEnabled) {
+                            onSuccess = ui.accessLater(
+                                    debugNotifier::publishResults, () -> {
+                                    });
+                        }
+                    }
+                    vaadinRequest.setAttribute(
+                            SERIALIZATION_TEST_REQUEST_ATTRIBUTE_KEY,
+                            new Runner(onSuccess));
+                });
+
+            } catch (Exception ex) {
+                LOGGER.debug("Error during serialization debug", ex);
+            }
+        }
+        return false;
+    }
+
+    static class Runner implements Consumer<HttpServletRequest> {
+
+        private final Consumer<Result> onSuccess;
+
+        public Runner(Consumer<Result> onSuccess) {
+            this.onSuccess = onSuccess;
+        }
+
+        private void executeOnSuccess(Result result) {
+            if (onSuccess != null) {
+                try {
+                    onSuccess.accept(result);
+                } catch (Exception ex) {
+                    // Do not interrupt the request
+                    ex.printStackTrace();
+                }
+            }
+        }
+
+        public void accept(HttpServletRequest request) {
+            HttpSession session = request.getSession(false);
+            if (session != null && request.isRequestedSessionIdValid()) {
+                serializeAndDeserialized(new WrappedHttpSession(session),
+                        this::executeOnSuccess);
+            }
+        }
+    }
+
+    public static class Filter extends HttpFilter {
+
+        private static final Logger LOGGER = LoggerFactory
+                .getLogger(Filter.class);
+
+        @Override
+        protected void doFilter(HttpServletRequest req, HttpServletResponse res,
+                FilterChain chain) throws IOException, ServletException {
+            chain.doFilter(req, res);
+            Object action = req
+                    .getAttribute(SERIALIZATION_TEST_REQUEST_ATTRIBUTE_KEY);
+            if (action instanceof Runner) {
+                LOGGER.debug(
+                        "Vaadin Request processed, Running Session Serialized Debug Tool");
+                ((Runner) action).accept(req);
+            }
+        }
+    }
+
+    public static void serializeAndDeserialized(WrappedSession session,
```

Comment:
Rename method to `serializeAndDeserialize`.

Reply:
Good catch!
Done